[Concept Entry] Python: os module (#6903)
Codecademy/docs: content/python/concepts/os-module/os-module.md

- `os.remove()`: Deletes a file from the filesystem
- `os.rmdir()`: Removes an empty directory
- `os.removedirs()`: Removes empty directories recursively
- `os.rename()`: Renames a file or directory
- `os.replace()`: Renames with cross-platform overwrite capability

## File/Directory Testing

The `os.path` submodule offers functions to test and validate file and directory paths, providing essential information about filesystem objects.

- `os.path.exists()`: Checks if a file or directory exists at the specified path
- `os.path.isfile()`: Determines if a path points to a file
- `os.path.isdir()`: Determines if a path points to a directory
- `os.path.getsize()`: Returns the size of a file in bytes
- `os.path.getmtime()`: Returns the last modification time of a file

## Environment Variables

The `os` module provides access to system environment variables, enabling programs to read configuration settings and system information.

- `os.environ`: A mapping object representing environment variables
- `os.getenv()`: Safely retrieves environment variables with optional default values
- `os.putenv()`: Sets environment variables for the current process

> > **Note:** Exercise caution when modifying environment variables as changes may affect system behavior and security.

Reviewer: ```suggestion
> **Note:** Be careful when changing environment variables, as these modifications can influence system behavior and security.
```
